server/services/prisonService.ts

```diff
@@ -0,0 +1,60 @@
+import type { Prison } from 'viewModels'
+import type { PrisonResponse } from 'prisonRegisterApiClient'
+import PrisonRegisterStore from '../data/cache/prisonRegisterStore'
+import PrisonRegisterClient from '../data/prisonRegisterClient'
+import toPrison from '../data/mappers/prisonMapper'
+import logger from '../../logger'
+
+const PRISON_CACHE_TTL_DAYS = 1
+
+export default class PrisonService {
+  constructor(
+    private readonly prisonRegisterStore: PrisonRegisterStore,
+    private readonly prisonRegisterClient: PrisonRegisterClient,
+  ) {}
+
+  async getPrisonByPrisonId(prisonId: string, token: string): Promise<Prison | undefined> {
+    const prisonResponse = await this.getPrison(prisonId, token)
+
+    if (prisonResponse) {
+      return toPrison(prisonResponse)
+    }
+
+    logger.info(`Could not find details for prison ${prisonId}`)
+    return undefined
+  }
+
+  private async getPrison(prisonId: string, token: string): Promise<PrisonResponse | undefined> {
+    return (
+      (await this.getCachedPrison(prisonId)) ||
```

Comment:
I missed the `||` initially. I may have been tempted to use an `if` statement, but this is fine 👍